src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1073,14 +1085,116 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed
+                        );
+                }
+            }
+
+            foreach ((string entityNameInMutation, HashSet<string> fieldsInEntity) in fieldsToAuthorize)
+            {
+                if (!AreFieldsAuthorizedForEntity(clientRole, entityNameInMutation, EntityActionOperation.Create, fieldsInEntity))
+                {
+                    throw new DataApiBuilderException(
+                        message: $"The client has insufficient permissions on one or more fields in the entity: {entityNameInMutation} referenced in this mutation.",
+                        statusCode: HttpStatusCode.Forbidden,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed
+                    );
+                }
+            }
+            
+            return fieldsToAuthorize;
+        }
+
+        private void PopulateMutationFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> fieldsToAuthorize,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters,
+            RuntimeConfig runtimeConfig)
+        {
+            InputObjectType schemaObject = ResolverMiddleware.InputObjectTypeFromIInputField(schema);
+            if (!fieldsToAuthorize.ContainsKey(entityName))
+            {
+                fieldsToAuthorize.Add(entityName, new HashSet<string>());
+            }
+
+            if (parameters is List<ObjectFieldNode> fields)
+            {
+                ProcessObjectFieldNodesForAuthZ(fieldsToAuthorize, entityName, context, runtimeConfig, schemaObject, fields);
             }
+            else if (parameters is ObjectValueNode objectValue)
+            {
+                ProcessObjectFieldNodesForAuthZ(fieldsToAuthorize, entityName, context, runtimeConfig, schemaObject, objectValue.Fields);
+            }
+            else if (parameters is List<IValueNode> values)
+            {
+                values.ForEach(value => PopulateMutationFieldsToAuthorize(fieldsToAuthorize, schema, entityName, context, value, runtimeConfig));
+            }
+        }
+
+        private void ProcessObjectFieldNodesForAuthZ(Dictionary<string, HashSet<string>> fieldsToAuthorize, string entityName, IMiddlewareContext context, RuntimeConfig runtimeConfig, InputObjectType schemaObject, IReadOnlyList<ObjectFieldNode> fieldNodes)
+        {
+            foreach (ObjectFieldNode field in fieldNodes)
+            {
+                Tuple<IValueNode?, SyntaxKind> fieldDetails = GetUnderlyingKindForField(field.Value, context.Variables);
+                SyntaxKind underlyingFieldKind = fieldDetails.Item2;
+                if (underlyingFieldKind != SyntaxKind.ObjectValue && underlyingFieldKind != SyntaxKind.ListValue)
+                {
+                    fieldsToAuthorize[entityName].Add(field.Name.Value);
```

Comment:
In case of M:N relationships, the fields from the linking table would also show up as a field in the related entity. Do we require a condition to check whether the field translates to a column in the related entity?

Reply:
Yep, that logic is yet to be added.

Reply:
Added